### DISCUSSION TITLE
特定メンバーのメッセージの一部項目が漏れているようです。

### DISCUSSION THREAD

**Discussion by kykskr** [proshunsuke/colmsg#72] [Q&A] [answered]
まず、こんな素敵なプログラムを作ってくださってありがとうございます。

教えていただいた方法を見てメッセージ抽出には成功しましたが、
特定メンバー（渡邉理佐）のメッセージ項目が一部漏れていることがわかりました。
グループ全体のメッセージを抽出していた時点でも、一人だけダウンロードが一度に完了できず、
同じコマンドを何回か追加入力して抽出しました。

現在、抽出されたメッセージ項目を確認した結果、
2019.02.13~2020.10.18
間のメッセージが全て漏れているようです。

このエラーについて何かアドバイスをお願いします🙇🏻‍♂️

**Comment by proshunsuke** [comment]
日本語で返信します。

すみません、返信が遅れました。

以下の項目を確認してください。

* 該当期間は購読していた期間でしょうか？購読していない期間のメッセージは取得出来ないです
* ファイルマネージャー(おそらくエクスプローラー)上のファイルのソートをファイル名順にして確認してください。 `_` で区切られる数字のうち先頭の数字が小さいほど昔のデータになります。漏れていないかもしれません
* `-d` オプションを使用してダウンロードフォルダを変更するか、現在保存されているフォルダを削除するなどして、最初から取得し直してみてください

**Reply by kykskr** [reply]
教えていただいた方法通りに何度も確認してみましたが、
特定区間のメッセージが漏れていることは確かだと思います。
特に見当がつく問題点がなくて推測が難しいですね..


親切に答えてくださってありがとうございます！
ずっと解決策を探してみます。

**Reply by proshunsuke** [reply]
例えばですが、画像にある `2019.02.13` と太字で示しているファイルの後ろは `2019.02.13` , `2019.02.14` , `2019.02.15` , ... `2019.03.09` と続いているように見えます。
もう一度確認していただけますか？

**Reply by kykskr** [reply]
あの当時、画像ファイルだけを対象に日付確認をしたので
きちんと確認できなかったようです。すいません🙇🏻







![220516](https://user-images.githubusercontent.com/104432965/168537040-3c4df249-47bc-451f-a855-fd7cfe223234.png)

しかし、教えていただいたとおりもう一度確認してみても
相変わらず違うところは発見できませんでした。
このメンバーだけcmd上から抽出が一度に完了しないという点も
関連があるのではないかと思って気になります。
私としてはどの部分で問題が発生したのか見当がつかないですね😅
(日本語が下手なのでちゃんと理解できない部分があるかもしれません。何卒ご了承ください。もし教えていただいた部分の中で、私が間違っている部分があれば直してくださるようお願いします。)



私の能力が足りないせいで何度も質問をして申し訳ありません。
気難しい質問なのに何度も答えてくださってありがとうございます🙇

**Reply by proshunsuke** [reply]
英語でも大丈夫ですよ

んー、なんでしょう...原因が分からないですね

`-F` オプションを使用することで、特定の日付以降のデータを指定して保存することが出来ます
`-d` オプションも使用して、以下のように実行して確認していただけますか？

```
colmsg --s_refresh_token <s_refresh_token> -n 渡邉理佐 -d path/to/temp/dir -F '2019/04/10 00:00:00'
```

**Reply by kykskr** [reply]
(返事が遅くなって、すみません。)

新しい解決策を提示してくださってありがとうございます！
幸いにも他の方が教えてくださった方法で問題を解決しました。
今まで一緒に悩んでくださって本当にありがとうございます。
全部皆さんの助けのおかげです🙇

**Comment by oidw** [comment, marked as answer]
こんにちは、すみません、私は日本語があまり上手ではないので、英語でお答えします。
Sorry, I'm not that good at Japanese. I'll reply in English.
I've run into a similar issue here. I don't know if it's caused by my unstable internet connection. Some data from one member was missed after I ran `colmsg --s_refresh_token <token>`.  I ran the same command several times after, but that part of data were still missing.
I'm not quite sure but I feels like that "colmsg" probably just checks the latest message data that've already been saved and would only save the new data after that, so I just removed the data after the 'break' (for your case, it might be the data after 20201013?) from the folder where I saved the data and ran the same command again, and I got all the data this time. 
I just want to share with you my similar experience if that helps, but I'm not sure if it is exactly the same issue as yours. I appologize for bothering you if it's completely irrevelent and hope you can solve this problem soon!

**Reply by kykskr** [reply]
(I'm not good at English, I hope you understand)

Sorry for the delayed reply.
Wow! Thank you so much. As you told me, when I delete data after the 'break' and re-entered the command, I succeeded in extracting the missing parts! thanks for your help, it's solved! thank you very much.
